specs.js

```diff
@@ -112,75 +135,571 @@ describe('when using trimperson api', () => {
+            .post('/CreateContainer', '*')
+            .query({securityToken: mockToken})
+            .reply(201, {
+              RecordNo: null
+            })
+      })
+      When(done => {
+        trim.createContainer(data, (err, body) => {
+          errReturn = err
+          dataReturn = body
+          done()
+        })
+      })
+      Then('An error is returned', done => {
+        expect(errReturn).to.exist
+        expect(dataReturn).not.to.exist
+        createContainerMock.done()
+        done()
+      })
+    })
+
+    describe('The container already exists', () => {
+      var createContainerMock, dataReturn, errReturn, data
+
+      Given(() => data = {
+        folderName: VALID_FOLDER_NAME,
+        privacyLevel: trim.PRIVACY_LEVELS.PUBLIC
+      })
+      Given(() => {
+        createContainerMock = nock(mockURL)
+            .filteringRequestBody(R.always('*'))
+            .post('/CreateContainer', '*')
+            .query({securityToken: mockToken})
+            .reply(204)
+      })
+      When(done => {
+        trim.createContainer(data, (err, body) => {
+          errReturn = err
+          dataReturn = body
+          done()
+        })
+      })
+      Then('An error is returned', done => {
         expect(errReturn).not.to.exist
         expect(dataReturn).to.exist
-        expect(dataReturn).to.be.a('string')
-        expect(dataReturn).to.equal('123456')
-        createRecordMock.done()
+        expect(dataReturn.RecordNo).to.equal(VALID_FOLDER_NAME)
+        createContainerMock.done()
+        done()
       })
+    })
+  })
 
+  /*
+   GET CONTAINER
+   */
+  describe('To get a container', () => {
+    Then('`getContainer` should be a function', (done) => {
+      expect(trim.getContainer).to.be.a('function')
+      done()
+    })
+    Then('`getPublicContainer` should be a function', (done) => {
+      expect(trim.getPublicContainer).to.be.a('function')
+      done()
+    })
+    Then('`getPrivateContainer` should be a function', (done) => {
+      expect(trim.getPrivateContainer).to.be.a('function')
+      done()
     })
 
-    describe('when the TRIM fails to recieve a createRecord request', () => {
-      var errReturn
-      var dataReturn
+    describe('If the public container is requested', () => {
+      var getContainerMock
       Given(() => {
-        createRecordMock = nock(mockURL)
-          .filteringRequestBody(function (bodyt) {
-            return '*'
-          })
-          .post('/AddRecordToTrim', '*')
-          .query({ securityToken: mockToken })
-          .reply(500, { msg: 'it failed wtfomg' })
+        getContainerMock = nock(mockURL)
+            .get('/GetContainer')
+            .query({trimid: VALID_CONTAINER, securityToken: mockToken})
+            .reply(200)
+      })
+      When(done => {
+        trim.getContainer(VALID_CONTAINER, trim.PRIVACY_LEVELS.PUBLIC, () => done())
+      })
+      Then('`/getContainer` is called', done => {
+        getContainerMock.done()
```

Comment:
`getContainerMock.done()` is the expectation. When you call done on the nock request, you get an error if the mocked request is not called. There were originally some comments regarding that, but I don't like comments that describe what a function call is doing - that's the domain of the documentation for the api you're invoking.